Update icons of the browser sidebar

hgiesel opened the pull request:
All icons of `SidebarItem`s are updated, except for flags.

Images:
<img width="216" alt="Screenshot 2021-06-16 at 16 23 07" src="https://user-images.githubusercontent.com/7188844/122240398-094dc100-cec2-11eb-8900-6dae088ff294.png">

<img width="218" alt="Screenshot 2021-06-16 at 16 24 59" src="https://user-images.githubusercontent.com/7188844/122240423-0eab0b80-cec2-11eb-8c3c-d7838e418f3e.png">

Highlights:
* I've added Notetype fields as `SidebarItem`s. These items will search for cards of this notetype, which are non-empty in this specific field. They also have a special menu item, to go to the Fields dialog. @RumovZ Could you possibly review d918a195856ca93c866af16bb55726c4e5b64de6?
* Collapse decks and tags have special icons.
* The current deck has a special icon
* Filtered decks have special icons
* The saved search icon now resembles a bookmark. There would also a heart and multiple heart icon available in the mdi iconset. 
* Flags and Card State have a neutral colour on their root icon, rather than grey.

Personally I like how the icons look more uniform now. 
I didn't change the flag icon, because the mdi version was a tad smaller, which made it hard to distinguish the colour, and ours doesn't stick out.
I'm not super happy with how the notetype icon and template/card type icon look a bit too similiar...

RumovZ commented:
That looks great! I've added one little comment.
I wouldn't worry too much about the template icon as it's hidden most of the time and hierarchy and identation also help to differntiate between notetypes and fields.

dae commented:
I like it as a whole; the previous tag icon stood out a bit compared to the existing other icons, and these mdi icons look better.

A couple of thoughts:

- Separate icons for decks with subdecks makes it feel a bit busy. There's already a disclosure icon denoting items with subitems - maybe that's enough?
- Same with the top level items - if we're showing multiples for the top level items then we probably should be doing it for all top-level items, and multiples for things like flags and today might look odd?
- Saved searches feels a bit similar to the deck icon - the advantage of the heart was it was more visually distinct

kleinerpirat commented:
After trying this out, I have to say this works really well. Bravo, Henrik!

> Saved searches feels a bit similar to the deck icon - the advantage of the heart was it was more visually distinct

@dae I would actually prefer the new bookmark icon because it represents the saving-functionality in a more literal way. Since it is all the way on top, I don't think the similarity to the deck icons will matter too much.

The resemblance between the deck and note icons (in comparison to the old icons) might be a bit more problematic. As they are quite small and rectangular in shape, they don't differ a lot at a quick glance:

![image](https://user-images.githubusercontent.com/62722460/122589841-0b01bb00-d061-11eb-9fb9-a971320c1cc0.png) ![image](https://user-images.githubusercontent.com/62722460/122589943-2a004d00-d061-11eb-84a3-2927a39a406e.png)


 Perhaps an icon that denotes HTML in some way would be more fitting:
`import { mdiCodeTags } from '@mdi/js';` ![code-tags](https://user-images.githubusercontent.com/62722460/122588940-f5d85c80-d05f-11eb-8a7b-aaaf3a265c63.png)

But I guess I'm biased and most users won't instantly connect note types with HTML.

dae commented:
I don't feel strongly about the saved searches icon, happy to go with whatever the majority likes

hgiesel commented:
I've removed the multiple-icons, and reverted to hearts for saved searches for now. I also couldn't unsee the resemblance to the bookmark in the deck icon. I think a star could also make sense, however that's also associated with marked cards. This could be a forum poll, but for now I'll stick to the heart.

I've adopted @kleinerpirat suggestion. After all, an "alternative" (old?) term for card types was "templates", so I've taken one that symbolizes this to some degree.

![Screenshot 2021-07-01 at 00 39 44](https://user-images.githubusercontent.com/7188844/124040438-35f6f200-da05-11eb-9aac-3bb4f5ccb841.png)

The `{}` here stands for `{{field references}}`.
At the same type I switched the note type icon. Being a black-on-white icon, and its frilly top edge makes it stand out a bit more.

To explain why I didn't choose other maybe _simpler_ icons, I already considered icons for other "objects" within Anki, and saved them for future use. I think the [card-bulleted-outline](https://materialdesignicons.com/icon/card-bulleted-outline), which I first used for card types, could symbolize cards, [note-outline](https://materialdesignicons.com/icon/note-outline) could symbolize notes, and [beaker-outline](https://materialdesignicons.com/icon/beaker-outline) could resemble deck options.

hgiesel commented:
I've reconsidered my previous judgement on the flag icons:
- By using the md icons, we also have access to the modifier versions, with little pluses or minuses, etc.
- For "no flag", I used the unfilled version. I think that's a nice way to symbolize "no color". (there's no striked through version, like the one for tags, which might have been even better).

<img width="204" alt="Screenshot 2021-07-01 at 01 22 05" src="https://user-images.githubusercontent.com/7188844/124043263-f3d0af00-da0a-11eb-8de6-553e55c71779.png">

dae commented:
The flag icon feels a bit pixelated to me, but the modifiers will presumably be useful, and I don't want to hold up the PR. Thanks!

hgiesel commented:
> The flag icon feels a bit pixelated to me.

That was exactly my thought as well. There's also `flag-alternative`, which is less pixelated, but has the above-mentioned issue of being smaller, so that the colours are harder to see (additionally there's no modifier versions).
